w2w/w2w.py

```diff
@@ -174,32 +186,103 @@ def main(argv=None):
+        .astype('int8')
+
+    return lcz_new
+
+def _check_lcz_wrf_extent(lcz, wrf) -> str:
 
     # Get bounding box coordinates
-    lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax = lcz.bounds
+    lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax = lcz.rio.bounds()
     wrf_xmin, wrf_ymin, wrf_xmax, wrf_ymax = \
         float(wrf.XLONG_M.min()), float(wrf.XLAT_M.min()), \
         float(wrf.XLONG_M.max()), float(wrf.XLAT_M.max())
 
-    # Evaluate and throw error if wrf not within LCZ domain
+    # Evaluate and throw error if WRF not within LCZ domain
     if not (wrf_xmin > lcz_xmin ) & (wrf_xmax < lcz_xmax ) & \
            (wrf_ymin > lcz_ymin) & (wrf_ymax < lcz_ymax):
 
-        print("ERROR: LCZ domain should be larger than WRF domain "
-              "in all directions.")
-        print(f"LCZ bounds  (xmin, ymin, xmax, ymax): "
-              f"{lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax}")
-        print(f"WRF bounds  (xmin, ymin, xmax, ymax): "
-              f"{wrf_xmin, wrf_ymin, wrf_xmax, wrf_ymax}")
+        message = "ERROR: LCZ domain should be larger than WRF domain " \
+                  "in all directions.\n" \
+                  "LCZ bounds (xmin, ymin, xmax, ymax): " \
+                 f"{lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax}\n" \
+                  "WRF bounds (xmin, ymin, xmax, ymax): "\
+                 f"{wrf_xmin, wrf_ymin, wrf_xmax, wrf_ymax}"
+    else:
+        message = "OK - LCZ domain is covering WRF domain"
+
+    return message
+
+
+def check_lcz_integrity(info: Dict[str, str], LCZ_BAND: int):
+
+    '''
+    Check integrity of LCZ GeoTIFF file, which should have:
+    - LCZ class labels between 1 and 17
+    - WGS84 (EPSG:4326) projection
+    - extent larger than WRF domain file, in all direction
+
+    Note that this procedure directly select correct band in GeoTIFF file,
+    so no need to select that specific band afterwards.
 
+    Output:
+        _clean.tif lcz file, used in the remainder of the tool.
+    '''
+
+    # Read the data
+    lcz = rxr.open_rasterio(info['src_file'])[LCZ_BAND, :, :]
+    wrf = xr.open_dataset(info['dst_file'])
+
+    # If any of [101, 102, 103, 104, 105, 106, 107] is in the lcz tif file.
+    lcz_100 = np.array(
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+         101, 102, 103, 104, 105, 106, 107]
+    )
+    if any(x in lcz.data.flatten() for x in lcz_100[10:]):
+        lcz = _replace_lcz_number(
+            lcz=lcz,
+            lcz_to_change=lcz_100,
+        )
+        print("LCZ class labels renamed from 1 to 17.")
+
+    # Re-project when not WGS84 (EPSG:4326)
+    if not lcz.rio.crs == CRS.from_epsg(4326):
+        lcz = lcz.rio.reproject("EPSG:4326")
+        lcz.data = xr.where(lcz.data > 0, lcz.data, np.nan)
+        print("LCZ map reprojected to WGS84 (EPSG:4326).")
+
+    # Check if LCZ map exceeds domain of geo_em file in all directions
+    message = _check_lcz_wrf_extent(lcz, wrf)
+    print(message)
+
+    if 'ERROR' in message:
         sys.exit()
```

Comment:
if we exit because of an error, we should always exit with a code or message, so it is != 0 (0 means success).

Reply:
exiting with 1 now.